tracing-subscriber/src/fmt/format.rs

```diff
@@ -292,6 +292,7 @@ impl<'a> fmt::Debug for Recorder<'a> {
 
 struct FmtCtx<'a, N> {
     ctx: &'a span::Context<'a, N>,
+    #[allow(unused)]
```

Comment:
The change is a bit more involving because I need to add more feature gates where the constructor for those structs are used. I put the diff in a gist here: https://gist.github.com/SnoozeTime/6d9b07c4d7ea9ef89ebcd4f1fa4b789d

What do you think about this change? If that is alright, I'll add it to the PR.

Reply:
I'm willing to go either way. I think in general, it's better to not compile in fields that are unused, but it is definitely a bigger change...